bin/run_OpenMP_VV.sh

```diff
@@ -103,129 +187,123 @@ if [ "$make_target" == "all" ] || [ "$custom_source" != "" ] ; then
+    perform_make OMP_VERSION=4.5
+    echo
+    echo "--------------------------- OMP 4.5 Detailed Results ---------------------------" >> combined-results.txt
+    echo "--------------------------- OMP 4.5 Results ---------------------------" > abrev.combined-results.txt
+    make_OpenMP_VV_reports 45
   fi
-fi
 
-if [ "$SKIP_SOLLVE51" != 1 ]; then
-  enable_xnack=0
-  if gpu_needs_xnack_for_usm "$AOMP_GPU" && ! is_apu && [ "$HSA_XNACK" == "" ]; then
-    export HSA_XNACK=1
-    enable_xnack=1
-    echo "Turning on HSA_XNACK=1 for 5.0 to allow USM tests to pass."
+  if [ "${SKIP_OPENMPVV_50}" != 1 ]; then
+    enable_xnack=0
+    if gpu_needs_xnack_for_usm "${AOMP_GPU}" && ! is_apu && [ "${HSA_XNACK}" == "" ]; then
+      export HSA_XNACK=1
+      enable_xnack=1
+      echo "Turning on HSA_XNACK=1 for 5.0 to allow USM tests to pass."
+    fi
+    # Run OpenMP 5.0 tests.
+    echo "--------------------------- START OMP 5.0 TESTING ---------------------"
+    export openmpvv_compile_flags="${openmpvv_compile_flags} -fopenmp-version=50"
+    perform_make OMP_VERSION=5.0 SOURCES="${openmpvv_test_filter}"
+    echo
+    echo "--------------------------- OMP 5.0 Detailed Results ---------------------------" >> combined-results.txt
+    echo "--------------------------- OMP 5.0 Results ---------------------------" >> abrev.combined-results.txt
+    make_OpenMP_VV_reports 50
+    if [ "${enable_xnack}" == 1 ]; then
+      unset HSA_XNACK
+    fi
   fi
-  echo "--------------------------- START OMP 5.1 TESTING ---------------------"
-  # Run OpenMP 5.1 Tests
-  export MY_SOLLVE_FLAGS="$MY_SOLLVE_FLAGS -fopenmp-version=51"
-
-# FIXME: Tests listed here are skipped to prevent GPUs crashing. Remove the test once issue is fixed.
-  custom_source="\" -type f ! \( -name *test_target_has_device_addr.c* \)\""
 
-  make tidy
-  make CC=$AOMP_BIN/clang CXX=$AOMP_BIN/clang++ FC=$AOMP_BIN/$FLANG CFLAGS="-lm $MY_SOLLVE_FLAGS" CXXFLAGS="$MY_SOLLVE_FLAGS" FFLAGS="$MY_SOLLVE_FLAGS"  OMP_VERSION=5.1 LOG=1 LOG_ALL=1 VERBOSE_TESTS=1 VERBOSE=1 SOURCES="$custom_source" all
-  echo
-  echo "--------------------------- OMP 5.1 Detailed Results ---------------------------" >> combined-results.txt
-  echo "--------------------------- OMP 5.1 Results ---------------------------" >> abrev.combined-results.txt
-  make_OpenMP_VV_reports 51
-  custom_source=""
-  if [ "$enable_xnack" == 1 ]; then
-    unset HSA_XNACK
+  if [ "${SKIP_OPENMPVV_51}" != 1 ]; then
+    enable_xnack=0
+    if gpu_needs_xnack_for_usm "${AOMP_GPU}" && ! is_apu && [ "${HSA_XNACK}" == "" ]; then
+      export HSA_XNACK=1
+      enable_xnack=1
+      echo "Turning on HSA_XNACK=1 for 5.1 to allow USM tests to pass."
+    fi
+    echo "--------------------------- START OMP 5.1 TESTING ---------------------"
+    # FIXME: Tests listed here are skipped to prevent GPUs crashing. Remove the test once issue is fixed.
+    openmpvv_test_filter="\" -type f ! \( -name *test_target_has_device_addr.c* \)\""
```

Comment:
During my testing, I noticed that this test works on `gfx90a`.
Does anybody know / remember under which circumstances this test caused issues?

Reply:
You could test on gfx908 and see if it works there. If so, let's enable and see if it fails again. (separate PR please)